src/python/pants/backend/python/subsystems/python_tool_base.py

```diff
@@ -23,6 +26,16 @@ class PythonToolRequirementsBase(Subsystem):
     default_interpreter_constraints: ClassVar[Sequence[str]] = []
     register_interpreter_constraints: ClassVar[bool] = False
 
+    # If this tool does not mix with user requirements (e.g. Flake8 and Isort, but not Pylint and
+    # Pytest), you should set this to True.
+    #
+    # You also need to subclass `PythonToolLockfileSentinel` and create a rule that goes from
+    # it -> PythonToolLockfileRequest by calling `PythonLockFileRequest.from_python_tool()`.
+    # Register the UnionRule.
+    register_lockfile: ClassVar[bool] = False
+    default_lockfile_resource: ClassVar[tuple[str, str] | None] = None
+    default_lockfile_file_path: ClassVar[str | None] = None
```

Comment:
Missed this the first time around, but: the fact that this gets baked into a Github URL will result in an incorrect URL if someone outside of the `pantsbuild/pants` repo declares a `PythonToolRequirementsBase` instance in a plugin.

Maybe make the field optional (even when `default_lockfile_resource` is set), and then declare it as:
```
    default_lockfile_file_url: ClassVar[str | None] = None
```
...and have instances in `pantsbuild/pants` call `git_url(cls.default_lockfile_file_path)` explicitly? Callers outside the repo could construct a URL or not if they didn't want to bother.

Reply:
True, good point. Will fix here.